server/routerlicious/packages/services-shared/src/healthCheckEndpoints.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Router } from "express";
+import { StartupChecker } from "./startupChecker";
+import { LumberEventName, Lumberjack } from "@fluidframework/server-services-telemetry";
+
+/**
+ * Checks if a service or functionality is ready for use.
+ * @internal
+ */
+export interface IReadinessCheck {
+	/**
+	 * Whether the service/functionality is ready for use.
+	 */
+	isReady(): Promise<boolean>;
+}
+
+/**
+ * Creates the health check endpoints for the service.
+ * @param serviceName - The name of the service.
+ * @param readinessCheck - Optional readiness check.
+ * @param createLivenessEndpoint - Whether to create the liveness endpoint. Services like Alfred already have a ping endpoint, so this can be set to false.
+ */
+export function createHealthCheckEndpoints(
+	serviceName: string,
+	readinessCheck?: IReadinessCheck,
+	createLivenessEndpoint = true,
+): Router {
+	const router: Router = Router();
+
+	const probeProps = {
+		serviceName,
+	};
+
+	router.get(
+		"/startup",
+		// eslint-disable-next-line @typescript-eslint/no-misused-promises
+		async (request, response) => {
+			const startupProbeMetric = Lumberjack.newLumberMetric(
+				LumberEventName.StartupProbe,
+				probeProps,
+			);
+			if (StartupChecker.getInstance().isStartupComplete()) {
+				response.sendStatus(200);
+			} else {
+				startupProbeMetric.error("Startup probe failed");
+				response.sendStatus(500);
+			}
+		},
+	);
+
+	if (createLivenessEndpoint) {
+		router.get(
+			"/ping",
+			// eslint-disable-next-line @typescript-eslint/no-misused-promises
+			async (request, response) => {
+				response.sendStatus(200);
+			},
+		);
+	}
+
+	router.get(
+		"/ready",
+		// eslint-disable-next-line @typescript-eslint/no-misused-promises
+		async (request, response) => {
+			const readinessProbeMetric = Lumberjack.newLumberMetric(
+				LumberEventName.ReadinessProbe,
+				probeProps,
+			);
+			if ((await readinessCheck?.isReady()) === false) {
```

Comment:
"router.get("/ready"..." route should probably be set only when readinessCheck has some value (it is not null and it is not undefined)

Reply:
For now, if the readiness provider implementation is not provided, the route returns a `OK 200`. This has been done to test all routes working fine even if we do not inject the implementation into OSS.